compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -365,21 +365,7 @@ object Applications {
 
   /** Splice new method reference `meth` into existing application `app` */
   private def spliceMeth(meth: Tree, app: Tree)(using Context): Tree = app match {
-    case Apply(fn, args) =>
-      // Constructors always have one leading non-implicit parameter list.
-      // Empty list is inserted for constructors where the first parameter list is implicit.
-      //
-      // Therefore, we need to ignore the first empty argument list.
-      // This is needed for the test tests/neg/i12344.scala
-      //
-      // see NamerOps.normalizeIfConstructor
-      //
-      if args == Nil
-         && !fn.isInstanceOf[Apply]
-         && app.tpe.isImplicitMethod
-         && fn.symbol.isConstructor
```

Comment:
I think that we should rather add another condition that checks if `meth` actually requires the empty argument list i.e.
```scala
&& !meth.tpe.widen.isNullaryMethod
```